dist/Scripts/ImportSQL.ps1

```diff
@@ -0,0 +1,155 @@
+      [Parameter(HelpMessage = "Wether SQL Server requires clients to specify HostNameInCertificate " + 
+         "rather than TrustServerCertificate=true. Defaults to false.")]
+      [switch]$ForceStrict
+   )
+ 
+   $ErrorActionPreference = "Stop"
+ 
+   # 1. Configure the Certificate, ForceEncryption and ForceStrict values that SQL Server should use
+   # Locate the "SuperSocketNetLib" registry key that contains the encryption settings; highest 
+   # first in case there are multiple versions.
+   $regKey = (ls "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SQL Server" -Recurse | 
+      Where-Object { $_.Name -like '*SuperSocketNetLib' } | Sort-Object -Property Name -Descending)
+   if($regKey -is [Array])
+   {
+      $regKey = $regKey[0]
+      Write-Warning "Multiple SQL instances found in the registry, using ""$($regKey.Name)""."
+   }
+ 
+   if($Certificate)
+   {
+      # The thumbprint must be in all lowercase, otherwise SQL server doesn't seem to accept it?!
+      Set-ItemProperty $regKey.PSPath -Name "Certificate" -Value $Certificate.Thumbprint.ToLowerInvariant()
+   }
+   else
+   {
+      Set-ItemProperty $regKey.PSPath -Name "Certificate" -Value $null
+   }
+       
+   # Note that ForceStrict is only available from SQL2022 onwards
+   Set-ItemProperty $regKey.PSPath -Name "ForceEncryption" `
+      -Value $(if($ForceEncryption) { 1 } else { 0 })
+   Set-ItemProperty $regKey.PSPath -Name "ForceStrict" `
+      -Value $(if($ForceStrict) { 1 } else { 0 })
+     
+   # 2. Grant READ access to the $Certificate's private key
+   if($Certificate)
+   {
+      # If no -ServiceAccount is specified, determine under which identity the MSSQLSERVER service 
+      # runs...
+      if(!$ServiceAccount)
+      {
+         $ServiceAccount = (Get-WmiObject Win32_Service -Filter "Name='MSSQLSERVER'").StartName
+      }
+       
+      # The private keys are stored on the file system, determine the path for $Certificate
+      # See https://stackoverflow.com/questions/40046916/how-to-grant-permission-to-user-on-certificate-private-key-using-powershell
+      if (!$Certificate.PrivateKey)
+      {
+          $privateKey = [System.Security.Cryptography.X509Certificates.RSACertificateExtensions]::GetRSAPrivateKey($Certificate)
+          $privateKeyPath = "$env:ProgramData\Microsoft\Crypto\Keys\$($privateKey.Key.UniqueName)"
+      }
+      else
+      {
+          $containerName = $Certificate.PrivateKey.CspKeyContainerInfo.UniqueKeyContainerName
+          $privateKeyPath = "$env:ProgramData\Microsoft\Crypto\RSA\MachineKeys\$containerName"
+      }
+       
+      # Now grant the $ServiceAccount read permissions to the private key file
+      $acl = Get-Acl -Path $privateKeyPath
+      $accessRule = New-Object System.Security.AccessControl.FileSystemAccessRule `
+         @($serviceaccount, "Read", "Allow")
+      [void]$acl.AddAccessRule($accessRule)
+      Set-Acl -Path $privateKeyPath -AclObject $acl
+   }
```

Comment:
I see what you're doing here. win-acme offers very similar functionality with the `--acl-fullcontrol` argument to set full control rights on the private key. I'm adding `--acl-read` in the next release (2.2.7) because this is the first time I've seen a use case for it, so you could do `--acl-read "NT Service\MSSQLSERVER"` for the next release and `--acl-fullcontrol "NT Service\MSSQLSERVER"` on the current releases if you prefer to take the native route (which will likely be better maintained, have better logs/error handling etc.).

The only thing that can't be done natively is this fancy bit of code:

`$ServiceAccount = (Get-WmiObject Win32_Service -Filter "Name='MSSQLSERVER'").StartName`

But you may script that part outside of the call to wacs.exe if you're unsure which account you're using. And I guess it's anyway not quite universal in it's current state due to named instances etc.

Reply:
I've removed this part and updated the example command in the comment section